presto-orc/src/main/java/com/facebook/presto/orc/OrcReader.java

```diff
@@ -166,17 +167,20 @@ public OrcReader(
             this.columnsToIntermediateKeys = ImmutableMap.of();
         }
 
-        try (InputStream metadataInputStream = new OrcInputStream(
+        Slice metadataSlice = orcFileTail.getMetadataSlice();
+        Optional<StripeMetaCache> stripeMetaCache = Optional.empty();
+        if (orcReaderOptions.isStripeMetaCacheEnabled()) {
```

Comment:
If StripeMetaCache is enabled and if the file does not support it, what happens ?

Reply:
The reader would proceed with reading index and footer from the file.
* For ORC files the Metadata object created by the OrcMetadataReader will have an empty Optional for the StripeMetaCache.
* In case of a DWRF file with an empty metadata slice the StripeMetaCache will be created, but it will be empty.

Reply:
> Also can we initialize StripeCache only if we have non-empty stripeCacheOffsets

done